Update README.md

TonyApuzzo opened the pull request:
Show how to ensure UTF16 output from kubectl is stored as ASCII so it will load into bash without errors.  NOTE: Used ASCII instead of UTF8 to avoid the Byte Order Mark (BOM) which can also cause issues when bash parses the file.

TonyApuzzo committed:
Update README.md

Show how to ensure UTF16 output from kubectl is stored as ASCII so it will load into bash without errors.  NOTE: Used ASCII instead of UTF8 to avoid the Byte Order Mark (BOM) which can also cause issues when bash parses the file.

tillig commented on README.md:
````diff
@@ -19,14 +19,17 @@ Commands like `kubectl` allow you to export command completion logic for use in
+    ```PowerShell
+    kubectl completion bash | Out-File -Encoding ASCII -NoNewline -FilePath kubectl_completions.sh
+    ```
+    **Make sure the completion file is UTF-8 or ASCII.** Some exports (like `kubectl`) come out as UTF-16. Windows `bash` may see this as a binary file that can't be interpreted which results in no completions happening.  Using the `Out-File` PowerShell cmdlet allows you to specify the encoding to ensure the completions script will load in bash.
````
If it really needs to be ASCII, this probably shouldn't be "UTF-8 or ASCII," right? Would you mind grabbing that one, too?

TonyApuzzo replied:
There are even more [complicated ways of creating UTF8 sans BOM](https://stackoverflow.com/questions/5596982/using-powershell-to-write-a-file-in-utf-8-without-the-bom) but those are kind of a pain when the command completions are safely ASCII.  In the future with PowerShell 6 we could switch to using `Set-Content` with UTF8NOBOM encoding.

TonyApuzzo committed:
Update README.md

Specify ASCII as the only acceptable encoding (for now) once PowerShell 6 become standard the `Set-Content` cmdlet will gain support for UTF8NOBOM encoding and that would be technically better.

TonyApuzzo committed:
Update README.md

Make more robust by ensuring that CRLF becomes LF as well as UCS2 becomes ASCII.

tillig commented:
This is awesome. Thank you so much for your help!
